### PR TITLE
SD-4329 WIP: Multi-region support for snowflake sso

### PR DESCRIPTION
**Pull request by scgkiran**

### CODE REVIEW

File: app/ui/setup.py [hunk 1]
```diff
@@ -172,3 +172,56 @@ def sndk_url(account: str, user: str, region: str) -> str:
...
+     api_allowed_prefixes = ({gateway_prefixes}) enabled = true;
+    GRANT USAGE ON INTEGRATION OPSCENTER_SUNDECK_EXTERNAL_FUNCTIONS TO APPLICATION "{app_name}";
+    CALL ADMIN.SETUP_EXTERNAL_FUNCTIONS();
+END;
+"""
+
+
+def get_api_gateway_prefixes(sf_region: str) -> str:
+    api_prefix_map = {
+        "us-east-1": "'https://1lf9af4dk7.execute-api.us-east-1.amazonaws.com', "
```
Comment (pravindra) [inline app/ui/setup.py]: can we avoid hardcoding the urls and the arn above ?

File: app/ui/setup.py [hunk 2]
```diff
@@ -172,3 +172,56 @@ def sndk_url(account: str, user: str, region: str) -> str:
...
+        + "'https://hh538sr9qg.execute-api.us-west-2.amazonaws.com', "
+        + "'https://w4cu711jd2.execute-api.us-west-2.amazonaws.com'",
+    }
+
+    return api_prefix_map[get_sndk_region(sf_region)]
+
+
+def get_sndk_region(sf_region: str) -> str:
+    # Supported Sundeck Regions ["us-east-1", "us-east-2.aws", "us-west-2"]
+    region_map = {
```
Comment (pravindra) [inline app/ui/setup.py:239]: so, it's us-west-2 for all the unsupported regions ?

File: app/ui/setup.py [hunk 3]
```diff
@@ -172,3 +172,56 @@ def sndk_url(account: str, user: str, region: str) -> str:
...
+        "us-east-2": "'https://mr2gl3hcuk.execute-api.us-east-2.amazonaws.com'",
+        "us-west-2": "'https://1fb567sika.execute-api.us-west-2.amazonaws.com', "
+        + "'https://hh538sr9qg.execute-api.us-west-2.amazonaws.com', "
+        + "'https://w4cu711jd2.execute-api.us-west-2.amazonaws.com'",
+    }
+
+    return api_prefix_map[get_sndk_region(sf_region)]
+
+
+def get_sndk_region(sf_region: str) -> str:
```
Comment (pravindra) [inline app/ui/setup.py]: can u pls call this get_sundeck_region ?

File: app/ui/setup.py (hunk 2 above)
Comment (scgkiran) [reply app/ui/setup.py:239]: yes I started with that, 
should we make all EU sf-regions use us-east-1 sundeck region.
We can leave all AP regions use us-west-2

Comment (pravindra) [reply app/ui/setup.py:239]: i am fine with this - it's the same thing we do for CNG too. i've updated the 1p also.

File: app/ui/setup.py (hunk 3 above)
Comment (scgkiran) [reply app/ui/setup.py]: done

File: app/ui/setup.py [hunk 4]
```diff
@@ -172,3 +173,80 @@ def sndk_url(account: str, user: str, region: str) -> str:
...
+    CREATE OR REPLACE API INTEGRATION OPSCENTER_SUNDECK_EXTERNAL_FUNCTIONS api_provider = aws_api_gateway """
+        + f""" api_aws_role_arn = '{OPSCENTER_ROLE_ARN}' api_allowed_prefixes = ({gateway_prefixes}) enabled = true;
+    GRANT USAGE ON INTEGRATION OPSCENTER_SUNDECK_EXTERNAL_FUNCTIONS TO APPLICATION "{app_name}";
+    CALL ADMIN.SETUP_EXTERNAL_FUNCTIONS();
+END;
+"""
+    )
+
+
+def get_redirect_url_for_security_integration(sf_region: str) -> str:
```
Comment (anshuldata) [inline app/ui/setup.py]: This method too should take deployment as parameter and hostname should be based on sf_region and deployment

prod --> https://api.sundeck.io/
stage --> https://api.stage.sndk.io/
dev --> https://api.dev.sndk.io/

File: deploy/deploy.py [hunk 5]
```diff
@@ -165,19 +171,23 @@ def main(argv):
...
+            print("==Deployment: ", deployment)
         elif opt == "-s":
             print("==Skipping Install")
             skip_install = True
         elif opt == "-x":
             print("==Skipping Package Creation")
             skip_package = True
 
-    execute(profile, version, not skip_install, skip_package)
+    execute(profile, version, deployment, skip_install, skip_package)
```
Comment (anshuldata) [inline deploy/deploy.py]: is "not" removal intentional ?

Comment (scgkiran) [reply deploy/deploy.py]: Fixed it

File: app/ui/setup.py (hunk 4 above)
Comment (scgkiran) [reply app/ui/setup.py]: Fixed it

File: app/ui/setup.py (hunk 2 above)
Comment (scgkiran) [reply app/ui/setup.py:239]: done

File: app/ui/setup.py (hunk 1 above)
Comment (scgkiran) [reply app/ui/setup.py]: done

**Review by anshuldata — APPROVED** [review]
lgtm

**Comment by scgkiran** [comment]
These changes are already merged into [PR](https://github.com/sundeck-io/OpsCenter/pull/51) on sundeck-io fork

**Pull request closed without merging**